components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/bapi/Axis2RFCHandler.java

```diff
@@ -70,34 +70,38 @@ public void handleRequest(JCoServerContext jCoServerContext,
         if (log.isDebugEnabled()) {
             log.debug("New BAPI function call received");
         }
-        workerPool.execute(new BAPIWorker(jCoServerContext, jCoFunction));
+        //TODO experimental code - validate
+        String xml = jCoFunction.toXML();
+        workerPool.execute(new BAPIWorker(xml));
     }
 
     private class BAPIWorker implements Runnable {
 
         private JCoServerContext serverContext;
         private JCoFunction function;
+        private String xmlContent;
 
         public BAPIWorker(JCoServerContext serverContext, JCoFunction function) {
             this.serverContext = serverContext;
             this.function = function;
         }
 
+        BAPIWorker(String xmlContent) {
+            this.xmlContent = xmlContent;
+        }
+
         public void run() {
             if (log.isDebugEnabled()) {
                 log.debug("Starting a new BAPI worker thread to process the incoming request");
             }
-            //TODO experimental code - validate
-            String xml = function.toXML();
-            ByteArrayInputStream bais = new ByteArrayInputStream(xml.getBytes());
+            ByteArrayInputStream bais = new ByteArrayInputStream(this.xmlContent.getBytes());
```

Comment:
ByteArrayInputStream implements Autocloseable shall we use try with resources for resource cleanup rather than closing the stream in the finally clause?

Reply:
+1